website/api/views.py

```diff
@@ -649,3 +654,95 @@ class CompanyViewSet(viewsets.ModelViewSet):
     filter_backends = (filters.SearchFilter,)
     search_fields = ("id", "name")
     http_method_names = ("get", "post", "put")
+
+
+class ContributorViewSet(viewsets.ModelViewSet):
+    queryset = Contributor.objects.all()
+    serializer_class = ContributorSerializer
+    permission_classes = (IsAuthenticatedOrReadOnly,)
+    http_method_names = ("get", "post", "put")
+
+
+class ProjectViewSet(viewsets.ModelViewSet):
+    queryset = Project.objects.all()
+    serializer_class = ProjectSerializer
+    permission_classes = (IsAuthenticatedOrReadOnly,)
+    http_method_names = ("get", "post", "put", "patch")
+
+    def create(self, request, *args, **kwargs):
+        data = request.data.copy()
+
+        name = data.get("name", "")
+        slug_base = slugify(name)
+        slug = slug_base
+        data["slug"] = slug
+
+        contributors = Project.get_contributors(self, data["github_url"])  # get contributors
+
+        project_instance = Project(
+            name=data["name"],
+            slug=slug,
+            description=data.get("description", ""),
+            github_url=data.get("github_url", ""),
+            wiki_url=data.get("wiki_url", ""),
+            homepage_url=data.get("homepage_url", ""),
+            logo=data.get("logo", ""),
+        )
```

Comment:
Shouldn't it be done through the serializer e.g. `serializer.is_valid()` and `serializer.save()`?